pkg/sdk/testint/event_tables_integration_test.go

```diff
@@ -0,0 +1,227 @@
+package testint
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/internal/collections"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/internal/random"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInt_EventTables(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	databaseTest, databaseCleanup := createDatabase(t, client)
+	t.Cleanup(databaseCleanup)
+	schemaTest, schemaCleanup := createSchema(t, client, databaseTest)
+	t.Cleanup(schemaCleanup)
+	tagTest, tagCleaup := createTag(t, client, databaseTest, schemaTest)
+	t.Cleanup(tagCleaup)
+
+	assertEventTableHandle := func(t *testing.T, et *sdk.EventTable, expectedName string, expectedComment string, expectedAllowedValues []string) {
+		t.Helper()
+		assert.NotEmpty(t, et.CreatedOn)
+		assert.Equal(t, expectedName, et.Name)
+		assert.Equal(t, "ACCOUNTADMIN", et.Owner)
+		assert.Equal(t, expectedComment, et.Comment)
+	}
+
+	cleanupTableHandle := func(t *testing.T, id sdk.SchemaObjectIdentifier) func() {
+		t.Helper()
+		return func() {
+			_, err := client.ExecForTests(ctx, fmt.Sprintf("DROP TABLE \"%s\".\"%s\".\"%s\"", id.DatabaseName(), id.SchemaName(), id.Name()))
```

Comment:
Should we have drop in the table or an event table should re-implement it ?

Reply:
i prefer to keep it as the API docs have it, but we could put some drop event table function which simply calls the drop table function in the background.